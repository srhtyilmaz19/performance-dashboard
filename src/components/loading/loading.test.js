import React from "react";

import { cleanup, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Loading from "./Loading";

afterEach(cleanup);

describe("<Loading />", () => {
  it("should render Loading component", function () {
    const { asFragment } = render(<Loading />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have class", function () {
    const { getByTestId } = render(<Loading />);

    const loadingWrapper = getByTestId("loading-wrapper");
    const loadingInner = getByTestId("loading-inner");

    expect(loadingWrapper).toHaveClass("loading-wrapper");
    expect(loadingInner).toHaveClass("lds-dual-ring");
  });

  it("should text class", function () {
    const { getByTestId } = render(<Loading />);

    const button = getByTestId("loading-text");

    expect(button).toHaveTextContent("Loading.");
  });
});
