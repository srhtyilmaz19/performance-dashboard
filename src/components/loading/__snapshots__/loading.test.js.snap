// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render Loading component 1`] = `
<DocumentFragment>
  <div
    class="loading-wrapper"
    data-testid="loading-wrapper"
  >
    <div
      class="lds-dual-ring"
      data-testid="loading-inner"
    />
    <span
      data-testid="loading-text"
    >
      Loading.
    </span>
  </div>
</DocumentFragment>
`;
